### PR TITLE
fix(client): prevent blank UI when LLM summary is non-string (#182)

### LINKED ISSUE #182: Broken UI when using Docker install
I setup pgEdge with Docker and installed LLM Proxy: ENABLED (provider: ollama, model: deepseek-r1:14b) - everything was up and running.

I click "Add Server" and the screen is blank (REACT error)

I add a server using curl (succeeded), I click on the link that showed 7 issues - (blank screen).

I then click on the server underneath (same server, but without the "7" icon overlay) - I get a blank screen.

I kill the tab and open another, but all I see now is a blank screen no matter what I do.

I've blown away my containers, images, and volumes multiple times - I always end up in a broken UI state.

Right now, this application is not ready for testing until these UI bugs and other issues are fixed.

Image

### PR DESCRIPTION
**Pull request by dpage**
## Summary

Closes #182. A user running the Docker compose stack with the LLM
proxy enabled (`provider: ollama`, `model: deepseek-r1:14b`) saw a
blank screen on every navigation, with two `React error #130` in the
console (production minified bundle: "objects are not valid as a React
child"). Once the first error fired, the previous error boundary
showed only a generic "Something went wrong" with no copy-pastable
diagnostic, so the screen stayed effectively blank across reloads and
container teardowns.

### Root cause

`<AIOverview>` is rendered for every selection (including the empty
welcome state when `aiEnabled`). When the LLM returned a structured
`summary` object — typical of reasoning models like deepseek-r1, which
emit `{thinking, answer}` shapes — the component rendered the object
directly as a React child via `{overview.summary}` and threw error
#130. Because no usable boundary surfaced the error, the user could
not even recognise the failure mode.

### Fixes

1. **`AIOverview` summary coercion** —
   `client/src/components/AIOverview/index.tsx` introduces a
   `summaryText` IIFE that coerces non-string summaries via
   `JSON.stringify` (with a `String(raw)` fallback for circular
   references) before they reach the renderer. The render site at line
   410 now uses `{summaryText}` instead of `{overview.summary}`.
2. **Top-level ErrorBoundary rewrite** —
   `client/src/components/ErrorBoundary.tsx` (already mounted in
   `App.tsx` wrapping `<AuthProvider><AppContent />`) now always shows
   the error message and component stack in a collapsible details
   block, regardless of `NODE_ENV`. A Reload button calls
   `window.location.reload()`. Errors are logged through the project
   `logger` instead of `console.error`. Optional `fallback` and
   `onError` props are accepted for future use.

### Tests

- `client/src/components/__tests__/ErrorBoundary.test.tsx` — new, 8
  tests, **100% line coverage** on the component.
- `client/src/components/__tests__/AIOverview.test.tsx` — extended
  with 26 tests covering `overviewUrl` branches, URL-change reset,
  fetch success/restart-detected/401/error paths, manual refresh,
  SSE-driven updates, collapse persistence including localStorage
  errors. Per-file line coverage of `AIOverview/index.tsx` rises from
  **79.36% to 98.41%**, well above the project's 90% floor.

### Verification

- [x] `make test-all` (collector, server, alerter, client) passes,
  exit 0.
- [x] `npm run lint` reports 0 errors (39 pre-existing warnings,
  none introduced).
- [x] `npm run build` succeeds.
- [x] Per-file line coverage on every touched file ≥90%.

## Test plan

- [ ] Reviewer runs the Docker compose stack with an LLM proxy
  configured for a reasoning model (e.g. ollama deepseek-r1:14b) and
  confirms the UI no longer blanks on navigation.
- [ ] Reviewer triggers a render error (e.g. via React DevTools) and
  confirms the boundary fallback renders the message, stack, and
  Reload button.
- [ ] CI passes on this PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented blank screens by coercing non-string summaries to safe text before rendering.
  * Redesigned error boundary to always show error message and collapsible details (default open) with a Reload action for recovery.

* **Documentation**
  * Added changelog entry describing the crash fix and error boundary improvements.

* **Tests**
  * Expanded coverage for summary coercion, error-boundary behavior, refresh/restart flows, polling/SSE edge cases, and UI interactions.

### CODE REVIEW

File: client/src/components/ErrorBoundary.tsx [hunk 1]
```diff
@@ -89,115 +125,150 @@ class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
...
 
-        // Store error info for display
+        // Persist the error info on state so the fallback UI can render
+        // the component stack.  React only passes errorInfo into
+        // componentDidCatch, never into getDerivedStateFromError.
         this.setState({ errorInfo });
 
-        // Call optional onError callback if provided
         if (this.props.onError) {
             this.props.onError(error, errorInfo);
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/ErrorBoundary.tsx]: Addressed in 4048862 — wrapped the callback in try/catch and route the secondary failure through `logger.error`. Added a test that asserts the fallback UI still renders when `onError` throws.

File: client/src/components/__tests__/ErrorBoundary.test.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            );
+
+            const reloadButton = screen.getByRole('button', { name: /reload/i });
+            fireEvent.click(reloadButton);
+            expect(reloadMock).toHaveBeenCalledTimes(1);
+
+            Object.defineProperty(window, 'location', {
+                configurable: true,
+                value: original,
+            });
```
[the remark this answers is not in the document]
Comment (dpage) [reply client/src/components/__tests__/ErrorBoundary.test.tsx]: Addressed in 9850cfa — wrapped the render/click/assert sequence in try/finally so `window.location` is restored even if an intermediate assertion throws.
